src/components/app/UrlManager.js

```diff
@@ -105,24 +145,33 @@ class UrlManager extends React.PureComponent<Props> {
   }
 
   render() {
-    const { isUrlSetupDone } = this.props;
-    return isUrlSetupDone ? (
-      this.props.children
-    ) : (
-      <div className="processingUrl" />
-    );
+    const { urlSetupPhase, children } = this.props;
+    switch (urlSetupPhase) {
+      case 'initial-load':
+        return <div className="processingUrl" />;
```

Comment:
Is this div ever actually presented to the user? It might be better to return null instead, since the div with a className implies some kind of presentation to the user. I'm guessing this pathway is completely synchronous.